daras_ai_v2/search_ref.py

```diff
@@ -302,3 +305,47 @@ def render_output_with_refs(state, height=500):
 def generate_footnote_symbol(idx: int) -> str:
     quotient, remainder = divmod(idx, len(FOOTNOTE_SYMBOLS))
     return FOOTNOTE_SYMBOLS[remainder] * (quotient + 1)
+
+
+#
+def generate_text_fragment(text, min_len=0, max_len=30):
+    """
+    Generate a text fragment for a given text.
+
+    Args:
+        text (str): The input text to generate a fragment from.
+        min_len (int): The minimum length for the prefix and suffix.
+        max_len (int): The maximum length for the prefix and suffix.
+
+    Returns:
+        str: The encoded text fragment.
+    """
+    def find_alphanumeric_window(s, window_size, reverse=False):
```

Comment:
This doesn't seem like it needs to be a locally scoped function, use top level scope

Reply:
yes, will do